packages/manager/apps/hub/src/app.module.js

```diff
@@ -0,0 +1,88 @@
+import 'script-loader!jquery'; // eslint-disable-line
+import 'script-loader!moment/min/moment-with-locales.min.js'; //eslint-disable-line
+import { Environment } from '@ovh-ux/manager-config';
+import angular from 'angular';
+import 'angular-translate';
+import uiRouter, { RejectType } from '@uirouter/angularjs';
+import ngOvhUiRouterLineProgress from '@ovh-ux/ng-ui-router-line-progress';
+
+import 'ovh-ui-angular';
+import ovhManagerCore from '@ovh-ux/manager-core';
+import ovhManagerHub from '@ovh-ux/manager-hub';
+import ovhManagerNavbar from '@ovh-ux/manager-navbar';
+import ovhManagerOrderTracking from '@ovh-ux/ng-ovh-order-tracking';
+import { detach as detachPreloader } from '@ovh-ux/manager-preloader';
+
+import get from 'lodash/get';
+import has from 'lodash/has';
+import head from 'lodash/head';
+
+import atInternet from './components/at-internet';
+import errorPage from './components/error-page';
+import catalog from './catalog';
+import dashboard from './dashboard';
+
+import routing from './routing';
+import './index.scss';
+import 'ovh-ui-kit/dist/oui.css';
+
+Environment.setVersion(__VERSION__);
+
+const moduleName = 'managerHubApp';
+
+angular
+  .module(moduleName, [
+    'pascalprecht.translate',
+    atInternet,
+    catalog,
+    dashboard,
+    errorPage,
+    ngOvhUiRouterLineProgress,
+    'pascalprecht.translate',
+    'oui',
+    ovhManagerCore,
+    ovhManagerHub,
+    ovhManagerNavbar,
+    ovhManagerOrderTracking,
+    'pascalprecht.translate',
+    uiRouter,
+  ])
+  .config(
+    /* @ngInject */ ($locationProvider) => $locationProvider.hashPrefix(''),
+  )
+  .config(routing)
+  .run(
+    /* @ngInject */ ($translate, $transitions) => {
+      $transitions.onBefore({ to: 'app.**' }, () => $translate.refresh());
+    },
+  )
+  .run(($translate) => {
+    moment.locale(head($translate.use().split('_')));
+  })
+  .run(
+    /* @ngInject */ ($state) => {
+      $state.defaultErrorHandler((error) => {
+        if (error.type === RejectType.ERROR) {
+          $state.go(
+            'error',
+            {
+              detail: {
+                message: get(error.detail, 'data.message'),
+                code: has(error.detail, 'headers')
+                  ? error.detail.headers('x-ovh-queryId')
+                  : null,
+              },
+            },
+            { location: false },
+          );
+        }
+      });
+    },
+  )
+  .run(/* @ngTranslationsInject:json ./translations */)
+  .run(
+    /* @ngInject */ ($rootScope, $transitions) =>
+      $transitions.onSuccess({}, () => detachPreloader()),
+  );
```

Comment:
This file was moved from previous `index.js`. I have only added this part to detach the preloader when the first state is reached.

Reply:
You need to call the hook only once no? Maybe something like that :
```
const unregisterHook = $transitions.onSuccess({}, () => {
  detachPreloader();
  unregisterHook();
})
```

Reply:
Sure, it was done like this before in angularJS to add class on preloader.

Reply:
Fixed.